upcloud/resource_upcloud_router_test.go

```diff
@@ -153,9 +286,72 @@ func testAccCheckRouterDestroy(s *terraform.State) error {
 	return nil
 }
 
+func testAccCheckRouterNetworkDestroy(s *terraform.State) error {
+	client := testAccProvider.Meta().(*service.Service)
+	for _, rs := range s.RootModule().Resources {
+		switch rs.Type {
+		case "upcloud_router":
+			routers, err := client.GetRouters()
+			if err != nil {
+				return fmt.Errorf("[WARN] Error listing routers when deleting upcloud router (%s): %s", rs.Primary.ID, err)
+			}
+
+			for _, router := range routers.Routers {
+				if router.UUID == rs.Primary.ID {
+					// service still found
+					return fmt.Errorf("[WARN] Tried deleting Router (%s), but was still found", rs.Primary.ID)
+				}
+			}
+		case "upcloud_network":
+			networks, err := client.GetNetworks()
+			if err != nil {
+				return fmt.Errorf("[WARN] Error listing networks when deleting upcloud network (%s): %s", rs.Primary.ID, err)
+			}
+
+			for _, network := range networks.Networks {
+				if network.UUID == rs.Primary.ID {
+					// service still found
+					return fmt.Errorf("[WARN] Tried deleting network (%s), but was still found", rs.Primary.ID)
+				}
+			}
+		}
+	}
+	return nil
+}
+
 func testAccRouterConfig(name string) string {
 	return fmt.Sprintf(`
 resource "upcloud_router" "my_example_router" {
   name = "%s"
 }`, name)
 }
+
+func testAccRouterNetworkConfig(routerName, networkName string, includeRouter, routerAttached bool) string {
```

Comment:
Not really fan of function that returns formated tf config, as the reader/reviewer cannot understand the test easly. up for debate

Reply:
Would you prefer full configs in the `Config`-field or what would be the preferred way of doing it? 

This way does have the advantage of cloning most of the configuration - eg. no need to change 3 configurations in the same way in three different steps but I do agree it's quite opaque and makes it quite tough to follow along, especially with the two bool parameters. Will rethink, at least, unless you have a great suggestion in mind?